modules/fedimint-ln-common/src/internal.rs

```diff
@@ -329,3 +306,45 @@ pub struct PreimageState {
 pub struct RefundSuccessState {
     refund_txid: TransactionId,
 }
+
+async fn fetch_and_validate_offer(
```

Comment:
I don't think this function and `create_incoming_contract_output` should be in the state machine code. I think it should be moved to lib.rs of ln-common

Reply:
ack. moved